.gitignore

```diff
@@ -0,0 +1,17 @@
+# OSX leaves these everywhere on SMB shares
+._*
+
+# OSX trash
+.DS_Store
+
+# Eclipse files
+.classpath
+.project
+.settings/**
+
+# Files generated by JetBrains IDEs, e.g. IntelliJ IDEA
+.idea/
+*.iml
+
+# Vscode files
+.vscode
```

Comment:
not sure why the gitignore is needed today, but this seems fine.

Reply:
+1 to adding the .gitignore, anything to help avoid accidental files added to the merge is great.